base/compiler/ssair/inlining.jl

```diff
@@ -1434,7 +1438,16 @@ function handle_const_prop_result!(
     return true
 end
 
-function concrete_result_item(result::ConcreteResult, state::InliningState)
+function handle_semi_concrete_result!(result::SemiConcreteResult, cases::Vector{InliningCase}, allow_abstract::Bool = false)
+    mi = result.mi
+    spec_types = mi.specTypes
+    allow_abstract || isdispatchtuple(spec_types) || return false
+    validate_sparams(mi.sparam_vals) || return false
+    push!(cases, InliningCase(spec_types, InliningTodo(mi, result.ir, result.effects)))
+    return true
+end
+
+function const_result_item(result::ConstResult, state::InliningState)
```

Comment:
Don't rename this function -- this is only called for `ConcreteResult`.
```suggestion
function concrete_result_item(result::ConcreteResult, state::InliningState)
```